joeynmt/training.py

```diff
@@ -149,6 +153,31 @@ def __init__(self, model: Model, config: dict) -> None:
             self.logger.info("Learning with token-level feedback.")
         self.return_logp = config["testing"].get("return_logp", False)
 
+    def _save_checkpoint_epochfreq(self) -> None:
+        """
+        Save the mode's current parameters and the training state to a
+        checkpoint by epoch_frequency
+
+        It does not follow the queue of _save_checkpoint
```

Comment:
Why not use the queue? This creates conflicting configurations. If there's a strong reason for skipping the queue, rather set the queue size to -1 at creation (when "save_freq" is > -1) and add a flag (e.g. "counts") and condition (`if counts`) to `_save_checkpoint` to avoid duplicated code.

Reply:
See above.